backend/danswer/connectors/freshdesk/connector.py

```diff
@@ -0,0 +1,144 @@
+import requests
+import json
+from datetime import datetime, timezone
+from typing import Any, List, Optional
+from danswer.file_processing.html_utils import parse_html_page_basic
+from danswer.configs.app_configs import INDEX_BATCH_SIZE
+from danswer.configs.constants import DocumentSource
+from danswer.connectors.interfaces import GenerateDocumentsOutput, PollConnector, LoadConnector, SecondsSinceUnixEpoch
+from danswer.connectors.models import ConnectorMissingCredentialError, Document, Section
+from danswer.utils.logger import setup_logger
+
+logger = setup_logger()
+
+
+class FreshdeskConnector(PollConnector, LoadConnector):
+    def __init__(self, batch_size: int = INDEX_BATCH_SIZE) -> None:
+        self.batch_size = batch_size
+
+    def ticket_link(self, tid: int) -> str:
+        return f"https://{self.domain}.freshdesk.com/helpdesk/tickets/{tid}"
+
+    def build_doc_sections_from_ticket(self, ticket: dict) -> List[Section]:
+        # Use list comprehension for building sections
```

Comment:
move this into the ticket->document function that you are going to make, its not worth being its own thing